### PR TITLE
19 cleanup

### LINKED ISSUE #19: Improperly forcing the use of IntentRequest
In alexawp.php, a new IntentRequest is instantiated within alexawp_news_request(), this should not always be an IntentRequest. Causes lots of errors in the log

### PR DESCRIPTION
**Pull request by tomharrigan**
Fixes #19 by using the Request class and then checking the Request type. Only IntentRequest has a name, so we get the name within that check. 

Adds a basic LaunchRequest handler for the News skill. 
Changes some code to use standard array syntax.
Fixes setting the `reason` property in SessionEndedRequest.
Adds StopIntent to intent schema

### CODE REVIEW

File: alexa-news.php [hunk 1]
```diff
@@ -14,97 +14,101 @@ class Alexa_News {
...
+					break;
+				case 'AMAZON.StopIntent':
+					$response->respond( __( 'Thanks for listening!', 'alexawp' ) )->endSession();
+					break;
+				default:
+					# code...
+					break;
+			}
+		} elseif ( $request instanceof Alexa\Request\LaunchRequest ) {
+			$response->respond( __( 'Ask me whats up!', 'alexawp' ) );
```
Comment (dlh01) [inline alexa-news.php]: Typo: `whats`.

File: alexa/request/certificate.php [hunk 2]
```diff
@@ -41,6 +41,10 @@ public function __construct( $certificate_url, $signature, $app_id = '' ) {
 	}
 
 	public function validate_request( $request_data ) {
+
+		// Set required http status code 400 for certificate error
+		http_response_code( 400 );
```
Comment (dlh01) [inline alexa/request/certificate.php]: Re. #27, `http_response_code()` is PHP 5.4+. Do you still want to use it?

File: fields.php [hunk 3]
```diff
@@ -221,6 +221,9 @@ function ( $taxonomy ) use ( $news_post_types ) {
 							],
 						],
 					],
+					[
+						'intent' => 'AMAZON.StopIntent',
+					]
```
Comment (dlh01) [inline fields.php]: Coding standards would suggest using a trailing comma.

**Review by dlh01 — COMMENTED** [review]
One question and a couple small things.

File: alexa/request/certificate.php (hunk 2 above)
Comment (tomharrigan) [reply alexa/request/certificate.php]: Are you ok with `header()` @dlh01 ?

Comment (dlh01) [reply alexa/request/certificate.php]: Sure. I only meant to mention it in light of the other issue; I defer to you about the minimum PHP version.

Comment (tomharrigan) [reply alexa/request/certificate.php]: Yea, I appreciate that, this is a good catch :)

Comment (tomharrigan) [reply alexa/request/certificate.php]: going with the wp status_header() function

**Pull request merged**